relayer/src/services/relayer/polygon-zkevm.ts

```diff
@@ -107,45 +122,73 @@ export class PolygonZKEvmRelayerService extends BaseRelayerService {
     }
   }
 
-  async prepare(): Promise<boolean> {
-    return true;
-  }
-
-  async relayTxToL1(l2TransactionHash: TransactionHash): Promise<string | undefined> {
-    console.log("\nClaiming PolygonZKEVM message on Ethereum L1.");
-
+  private async getRelayBridgeEventParameters(
+    l2TransactionHash: TransactionHash,
+  ): Promise<BridgeEventData> {
     const receipt = await this.l2RpcProvider.waitForTransaction(l2TransactionHash, 1, 300000);
 
     this.checkTransactionValidity(receipt, "Polygon ZKEvm");
 
     // 1. Extract bridge event parameters required for claiming a message
-    const bridgeEventParameters = await parseBridgeEvent(receipt.logs);
+    const bridgeEventParameters = parseBridgeEvent(receipt.logs);
     if (!bridgeEventParameters) {
       throw new Error(
         `Cannot find an event ("BridgeEvent") in the L2 transaction ${l2TransactionHash}`,
       );
     }
     console.log("Found BridgeEvent. Proceeding with the next steps...");
+    return bridgeEventParameters;
+  }
 
-    // 2. Fetch message data
-    const message = await this.getMessageInfoSafe(
-      bridgeEventParameters.depositCount,
-      bridgeEventParameters.originNetwork,
-    );
+  private async isRelayCompleted(
+    l2TransactionHash: TransactionHash,
+  ): Promise<TransactionHash | false> {
+    if (!this.relayBridgeEventParameters) {
+      this.relayBridgeEventParameters = await this.getRelayBridgeEventParameters(
```

Comment:
Same here as in `arbitrum.ts` service. Declare the variable inside the scope of this function instead.

Reply:
:heavy_check_mark: